### ISSUE TITLE
SpamAssassin index error during parsing body details

### ISSUE THREAD

**Issue by danieleperera** [ninoseki/eml_analyzer#229]
I'm encountering a parsing error at line:
https://github.com/ninoseki/eml_analyzer/blob/a756fdbfb6f7553b8233e486a33b9de375438c22/backend/clients/spamassasin.py#L47

When I've tested the code with the following sample email:
```
GTUBE = """Subject: Test spam mail (GTUBE)
Message-ID: <GTUBE1.1010101@example.net>
Date: Wed, 23 Jul 2003 23:30:00 +0200
From: Sender <sender@example.net>
To: Recipient <recipient@example.net>
Precedence: junk
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Transfer-Encoding: 7bit

--- This message is from an EXTERNAL SENDER - be CAUTIOUS, particularly with links and attachments! --- 

This is the GTUBE, the
    Generic
    Test for
    Unsolicited
    Bulk
    Email

If your spam filter supports it, the GTUBE provides a test by which you
can verify that the filter is installed correctly and is detecting incoming
spam. You can send yourself a test mail containing the following string of
characters (in upper case and with no white spaces and line breaks):

XJS*C4JDBQADN1.NSBN3*2IDNEN*GTUBE-STANDARD-ANTI-UBE-TEST-EMAIL*C.34X

You should send this test mail from an account outside of your network.
""".encode(
    "ascii"
)
```

The code breaks as the sample contains `--- This message is from an EXTERNAL SENDER - be CAUTIOUS, particularly with links and attachments! --- ` in the body. 

Using `----` instead of `---` resolves the issue.

```python
for index, line in enumerate(lines):
    if "----" in line:
        delimiter_index = index + 1
        break
```
